src/riak_core_info_service_process.erl

```diff
@@ -0,0 +1,69 @@
+-module(riak_core_info_service_process).
+-behaviour(gen_server).
+
+-record(state, {
+    source = undefined :: riak_core_info_service:callback(),
```

Comment:
yea - go with `service_provider` unless you come up with something better - please make sure to rename things like the variables in `riak_core_app.erl` to match whatever you come up with.